### PR TITLE
XWIKI-17619: Mention queue locking exception

### PR DESCRIPTION
**Pull request by manuelleduc**
Jira: https://jira.xwiki.org/browse/XWIKI-17619

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/async/DefaultMentionsBlockingQueueProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import ch.rasc.xodusqueue.XodusBlockingQueue;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test of {@link DefaultMentionsBlockingQueueProvider}.
+ *
+ * @version $Id$
+ * @since X.Y.Z
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/async/DefaultMentionsBlockingQueueProviderTest.java:45]: Missing info.

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/async/DefaultMentionsBlockingQueueProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    @MockComponent
+    private Environment environment;
+
+    private File tempDir;
+
+    @BeforeEach
+    void setUp() throws IOException
+    {
+        this.tempDir = Files.createTempDirectory("DefaultMentionsBlockingQueueProviderTest").toFile();
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/async/DefaultMentionsBlockingQueueProviderTest.java:61]: We have best practices for temporary dir in test: https://dev.xwiki.org/xwiki/bin/view/Community/Testing/JavaUnitTesting/#HTemporaryDirectory

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/async/DefaultMentionsBlockingQueueProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import static org.mockito.Mockito.when;
+
+/**
+ * Test of {@link DefaultMentionsBlockingQueueProvider}.
+ *
+ * @version $Id$
+ * @since X.Y.Z
+ */
+@ComponentTest
+class DefaultMentionsBlockingQueueProviderTest
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/async/DefaultMentionsBlockingQueueProviderTest.java:48]: You're missing a test to check closeQueue no?

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/async/DefaultMentionsBlockingQueueProviderTest.java:48]: I'm testing it indirectly with `initBlockingQueueTwice`.

Comment (surli) [reply xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/async/DefaultMentionsBlockingQueueProviderTest.java:48]: Hm right it's a real implem not a mock so you cannot assess the behaviour. Ok then.

**Pull request merged**